### ISSUE TITLE
No matching static site was found or the api key was invalid.

### ISSUE THREAD

**Issue by theeKatlego** [Azure/static-web-apps#198]
Hi, I get the following error 

`The content server has rejected the request with: BadRequest
Reason: No matching static site was found or the api key was invalid.`

I've changed the structure of the repo by moving the project into a subfolder. I then updated the workflow. I did not make any changes on the static web app on Azure.
[rickvandenbosch](https://www.rickvandenbosch.net/blog/azure-static-web-apps-quirks-gotchas/) says this also happens when changing branch names. Note that I did not change the branch name.

**Workflow file**
```
name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - master
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - master

jobs:        
  build-and-deploy:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy
    steps:
    - uses: actions/checkout@v2
      with:
        submodules: true
    - uses: actions/setup-node@v1
      with:
        node-version: 12.x
        registry-url: https://registry.npmjs.org/
    - name: yarn install and build
      run: |
        cd zoza-communications
        yarn install
        yarn build    
    - name: Deploy
      id: deploy
      uses: Azure/static-web-apps-deploy@v0.0.1-preview
      with:
        azure_static_web_apps_api_token: ${{ ***** }}
        repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
        action: "upload"
        ###### Repository/Build Configurations - These values can be configured to match you app requirements. ######
        # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
        app_location: "zoza-communications" # App source code path
        api_location: "api" # Api source code path - optional
        app_artifact_location: "dist/" # Built app content directory - optional
        ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v0.0.1-preview
        with:
          azure_static_web_apps_api_token: ${{ **** }}
          action: "close"
```

repo - https://github.com/theeKatlego/zoza-communications

**Comment by miwebst** [comment]
Looks like you have 3 GitHub actions configured against the master branch of your repo, is that desired?

https://github.com/theeKatlego/zoza-communications/tree/master/.github/workflows

My guess is a couple of these are leftover from previous static web apps that have since been deleted.

**Comment by theeKatlego** [comment]
Yes, I have 3 against the master branch of your repo. There are 2 web apps in the repo.

`azure-static-web-apps-salmon-rock` was added as a test. I created a temp Azure Static Web App and deployed to it successfully. You can ignore this.

Initially, I just had `azure-static-web-apps-wonderful-sky` and it worked fine. I then did the following 

1. Moved `wonderful-sky` web app source code in my repo to a subfolder - `./zoza-communications`
2. Added `wonderful-tree` web app source code in my repo to a subfolder - `./zoza-communications-portal`
3. Created Azure Static Web App on Azure Portal. This created `azure-static-web-apps-wonderful-tree` workflow.
4. Lastly, I updated the `azure-static-web-apps-wonderful-sky` since I've moved the source code.

`wonderful-tree` deploys fine but it seems I managed to break `wonderful-sky`.

I also copied `wonderful-sky` jobs to `wonderful-tree`. Jobs to deploy to `wonderful-sky` still fail.

Ideally, I want to have one action to deploy both `wonderful-sky` and `wonderful-tree`.

**Comment by miwebst** [comment]
I see. Having one GitHub action workflow file to deploy both will be challenging since each workflow file references the deployment secret of a particular static web app. You can definitely have 2 workflow files so a single commit to master will deploy both.

As for the issue that you seem to be hitting, on closer inspection of https://github.com/theeKatlego/zoza-communications/blob/master/.github/workflows/azure-static-web-apps-wonderful-tree-0adc4b81e.yml, it appears to be using the  api key (deployment secret) from the WONDERFUL_SKY_0EC6BE903 static web app. If you change this back to be the api key that was generated for wonderful-tree that should fix things!

**Comment by theeKatlego** [comment]
I don't quite understand why `wonderful-sky` secret isn't valid anymore. The `wonderful-sky` workflow worked fine with its respective secret before. What could've made it invalid? I did not update it.

So, I made the change as per your suggestion. Site deployed to `wonderful-tree` fine. However, I have two Azure Static Web App `wonderful-sky` and `wonderful-tree`. Each site has its respective workflow. 

I've now separated the workflows and each references its respective static web app secret.
`wonderful-tree` still works fine.
`wonderful-sky` fails with the following error.
```
The content server has rejected the request with: BadRequest
Reason: No matching static site was found or the api key was invalid.
```

Is there a way to generate a new key or get the current one from the portal?
